hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/oauth2/AzureADAuthenticator.java

```diff
@@ -109,15 +112,12 @@ public static AzureADToken getTokenUsingClientCreds(String authEndpoint,
    * @return {@link AzureADToken} obtained using the creds
    * @throws IOException throws IOException if there is a failure in obtaining the token
    */
-  public static AzureADToken getTokenFromMsi(String tenantGuid, String clientId,
+  public static AzureADToken getTokenFromMsi(final String authEndpoint, String tenantGuid, String clientId,
                                              boolean bypassCache) throws IOException {
-    String authEndpoint = "http://169.254.169.254/metadata/identity/oauth2/token";
-
     QueryParams qp = new QueryParams();
     qp.add("api-version", "2018-02-01");
     qp.add("resource", RESOURCE_NAME);
 
-
     if (tenantGuid != null && tenantGuid.length() > 0) {
       String authority = "https://login.microsoftonline.com/" + tenantGuid;
```

Comment:
If you are tuning the endpints, I think this is the ideal place to also address the "no login in Azure china" issue

Reply:
value for authority also made configurable

Reply:
thanks